src/transactions/test/PathPaymentTests.cpp

```diff
@@ -112,6 +112,78 @@ TEST_CASE("pathpayment", "[tx][pathpayment]")
 
     closeLedgerOn(*app, 2, 1, 1, 2016);
 
+    SECTION("transact more than INT64_MAX in a path payment")
+    {
+        auto a1 = root.create("A1", minBalance4);
+        auto a2 = root.create("A2", minBalance4);
+        auto a3 = root.create("A3", minBalance4);
+
+        a1.changeTrust(idr, trustLineLimit);
+        a1.changeTrust(cur2, trustLineLimit);
+
+        a2.changeTrust(cur2, trustLineLimit);
+        a2.changeTrust(cur1, trustLineLimit);
+
+        a3.changeTrust(idr, trustLineLimit);
+        a3.changeTrust(cur2, trustLineLimit);
+        a3.changeTrust(cur1, trustLineLimit);
+
+        SECTION("issue more than INT64_MAX")
+        {
+            for_versions_from(11, *app, [&] {
+                gateway.pay(a1, idr, trustLineLimit);
+                gateway.pay(a2, cur1, trustLineLimit);
+                gateway.pay(a3, cur2, trustLineLimit);
+
+                a1.manageBuyOffer(0, idr, cur2, Price{1, 1}, INT64_MAX,
+                                  MANAGE_OFFER_CREATED);
+
+                a2.manageBuyOffer(0, cur1, cur2, Price{1, INT32_MAX}, INT64_MAX,
+                                  MANAGE_OFFER_CREATED);
+
+                a3.manageOffer(0, cur2, idr, Price{2, 1}, INT64_MAX,
+                               MANAGE_OFFER_CREATED);
+
+                REQUIRE(a1.getTrustlineBalance(cur2) == 0);
+                REQUIRE(a2.getTrustlineBalance(cur2) == 0);
+
+                // CUR2 -> IDR -> CUR2 -> CUR1
+                gateway.pay(a3, cur2, INT64_MAX, cur1, INT32_MAX, {idr, cur2});
+
+                REQUIRE(INT64_MAX - a1.getTrustlineBalance(cur2) <
+                        a2.getTrustlineBalance(cur2));
+            });
+        }
+
+        SECTION("burn more than INT64_MAX")
```

Comment:
Maybe I'm missing something, but this also doesn't seem to be burning more than `INT64_MAX`? I see a similar situation to the other section.

Reply:
Good catch. I was looking at the test as the issuer sending/receiving over INT64_MAX, but I didn't take into account the intermediate hop that did the opposite. After thinking about this, I realized I was making the test much more complicated than it needed to be. The updated test uses a path payment from `a1` to hit offers set up by the issuer.